File: multi-llm-inference.js
const url = "<Request url from team APEX:E3>";

// Supported model ids
// gemini-1.5-flash // gemini-2.0-flash // gpt-4o-turbo // o1-mini // o3-mini // o1
// claude-sonnet-3.5 // mistral-large-latest // grok2 // deepseek // nemotron

const body = {
  "prompt": `<your prompt>`,
  "model": "<moodel id>", 
  "systemPrompt": "Your job is to summarise earnings transcripts",
  "maxOutputTokens": 1000,
  "userId": "<your-email>"
};

fetch(url, {
  method: 'POST',
  headers: {
    'Content-Type': 'application/json'
  },
  body: JSON.stringify(body)
})
.then(response => {
  if (response.ok) {
    if (response.status === 201) {
      return response.text().then(text => console.log("Response:", text));
    } else {
      console.log("Unexpected success status code:", response.status);
    }
  } else {
    console.log("Failed to get a response. Status code:", response.status);
  }
})
.catch(error => console.error('Fetch error:', error));

File: multi-llm-inference.py
import requests

# Supported model ids
# gemini-1.5-flash // gemini-2.0-flash // gpt-4o-turbo // o1-mini // o3-mini // o1
# claude-sonnet-3.5 // mistral-large-latest // grok2 // deepseek // nemotron
# gpt-5-chat-latest 
# grok-4
# private Qwen/Qwen2.5-VL-72B-Instruct
# gemini-2.5-pro
# claude-sonnet-4
# o3


def main():
    url = "<Request url from team APEX:E3>"

    body = {
        "prompt": "<your prompt>",
        "model": "<model id>",  
        "systemPrompt": "Your job is to answer questions precisely.",
        "maxOutputTokens": 1000,
        "userId": "<your-email>"
    }

    headers = {
        "Content-Type": "application/json"
    }

    try:
        response = requests.post(url, json=body, headers=headers)
        # Check if the request was successful
        if response.ok:
            # Check if the status code is exactly 201
            if response.status_code == 201:
                print("Response:", response.text)
            else:
                print("Unexpected success status code:", response.status_code)
        else:
            print("Failed to get a response. Status code:", response.status_code)
    except requests.exceptions.RequestException as e:
        print("Fetch error:", e)

if __name__ == "__main__":
    main()
